lesson_5_hw.js

```diff
@@ -0,0 +1,68 @@
+/*
+1. 
+  Напиши функцію map(fn, array), яка приймає на вхід функцію та масив, 
+  та обробляє кожен елемент масиву цією функцією, повертаючи новий масив. 
+  Приклад:
+*/
+
+function map(fn, array) {
+  // ваш код
+  return array.map(fn);
+}
+
+function square(x) {
+  return x * x;
+} // повертає квадрат числа
+
+console.log(map(square, [1, 2, 3, 4])); // [1, 4, 9, 16]
+console.log(map(square, [])); // []
+
+// Зверніть увагу: функція не повинна змінювати переданий їй масив:
+
+var arr = [1, 2, 3];
+console.log(map(square, arr)); // [1, 4, 9]
+console.log(arr); // [1, 2, 3]
+
+/*
+  2. Написати функцію яка форматує текст: 
+    - прибирає пробіли з початку і кінця, 
+    - прибарає всі займі пробіли (два і більше підряд)
+    - кожне речення починає з великої букви, а всі наступні символи в речені робить маленькими
+*/
+function formatText(text) {
+  return text
+      .replace(/\s+/g, ' ')
+      .replace(/\. /g, ". ")
+      .toLowerCase()
+      .replace(/(^|\.)\s*([a-z])/g, (match, p1, p2) => p1 + p2.toUpperCase());
```

Comment:
```suggestion
      .replace(/(^|\.)\s*([a-z])/g, (match, p1, p2) => p1 + ' ' + p2.toUpperCase()).trim();
```
Відсутній пробіл між реченнями